sql_generators/glean_usage/common.py

```diff
@@ -223,17 +237,31 @@ def generate_per_app_id(
             return
 
         if output_dir:
-            write_sql(
-                output_dir, view, "metadata.yaml", view_metadata, skip_existing=True
-            )
-            write_sql(output_dir, view, "view.sql", view_sql, skip_existing=True)
-            write_sql(
-                output_dir, table, "metadata.yaml", table_metadata, skip_existing=True
-            )
-            write_sql(output_dir, table, "query.sql", query_sql, skip_existing=True)
+            # generated files to update
+            Artifact = namedtuple("Artifact", "table_id basename sql")
+            artifacts = [
+                Artifact(view, "metadata.yaml", view_metadata),
+                Artifact(view, "view.sql", view_sql),
+                Artifact(table, "metadata.yaml", table_metadata),
+                Artifact(table, "query.sql", query_sql),
+            ]
 
             if not self.no_init:
-                write_sql(output_dir, table, "init.sql", init_sql, skip_existing=True)
+                artifacts.append(Artifact(table, "init.sql", init_sql))
+
+            for artifact in artifacts:
+                destination = (
+                    get_table_dir(output_dir, artifact.table_id) / artifact.basename
+                )
+                skip_existing = destination in self.override_skip()
```

Comment:
Calling `self.override_skip()` within a loop is less than ideal.  I'd suggest either saving the value in a variable outside the loop, or memoizing the method so it doesn't recalculate the skip list on every call.